meson.build

```diff
@@ -122,15 +121,23 @@ if cxx.has_function('sigwait')
 endif
 
 
-# XXX - why do we have to define this manually?
-if cxx.has_header('libintl.h')
-  add_project_arguments('-DHAVE_GETTEXT', language: ['c', 'cpp'])
-
-  if have_darwin or have_freebsd or have_windows
-    add_project_link_arguments('-lintl', language: ['c', 'cpp'])
+if meson.version().version_compare('>= 0.59')
+  intl_dep = dependency('intl', required: false)
+  intl_found = intl_dep.found()
+else
+  if cxx.has_function('ngettext')
```

Comment:
This is insufficient, since it will test whether ngettext is present in libc.

However, if it is both present in libc and present as a standalone library for the GNU libintl port, has_function will return true, but attempting to include libintl.h will cause source code referring to `ngettext` to compile into object code referencing `libintl_ngettext`.

You need to use the `prefix:` kwarg to has_function in order to ensure that the generated test code uses the *right* libintl.h; the builtin dependency handles this for you.

Reply:
You also need to avoid has_function entirely if you want to correctly handle some obscure edge cases like "works on mingw", see https://github.com/mesonbuild/meson/commit/1ac6fc0595f992621cbb79b7971ba07db1fa8e62.

Reply:
I have added the `prefix:` kwargs as suggested, this should cover all platforms we care about. And it's only relevant for the fallback code anyway. Most users should have Meson 0.59 or later meanwhile.

Question: Does it make sense to use `has_headers: 'libintl.h'` also for `cxx.find_library()`?

Thanks for the review and implementing the builtin dependencies upstream. :+1:

Reply:
find_library doesn't generate prototype code (other than `int main() { return 0; }` at least) and do a linkability test. A header cannot influence whether the library exists, the way it can influence whether a symbol exists.

Instead, the has_headers kwarg for find_library is a way of succinctly checking for two things in a single pass, with included short-circuiting if the first thing checkedreturns notfound. In order:
- cxx.has_header()
- cxx.find_library()

I would not bother checking for it, because my assumption is that if a third-party library exists then so do its headers.